docs/core/tutorials/libraries.md

```diff
@@ -235,236 +186,126 @@ And in the middle of the source, you can use `#if` directives to use those libra
-
-```json
-{
-    "frameworks":{
-        "netstandard1.6":{
-           "dependencies":{
-                "NETStandard.Library":"1.6.0",
-            }
-        },
-        ".NETPortable,Version=v4.0,Profile=Profile328":{
-            "buildOptions": {
-                "define": [ "PORTABLE328" ]
-            },
-            "frameworkAssemblies":{
-                "mscorlib":"",
-                "System":"",
-                "System.Core":"",
-                "System.Net"
-            }
-        }
+             return $"dotnetfoundation.org mentions .NET {dotNetCount} times in its HTML!";
+         }
+ #endif
     }
 }
-
 ```
 
-Now you can conditionally compile against that target:
+If you build this project with `dotnet build`, you'll notice three directories under the `bin/` folder:
 
-```csharp
-#if !PORTABLE328
-using System.Net.Http;
-using System.Threading.Tasks;
-// Potentially other namespaces which aren't compatible with Profile 328
-#endif
 ```
-
-Because `PORTABLE328` is now recognized by the compiler, the PCL Profile 328 library generated by a compiler will not include `System.Net.Http` or `System.Threading.Tasks`.
-
-When you build a project that includes PCL Profile 328 and `netstandard1.6` as frameworks in the *project.json* file, it will have these subfolders in the */bin/debug* folder:
-
-```
-portable-net40+sl50+netcore45+wpa81+wp8/
-netstandard1.6/
+net40/
+net45/
+netstandard1.4/
 ```
 
-## How to use native dependencies
+Each of these contain the `.dll` files for each target.
 
-You may wish to write a library which depends on a native `.dll` file.  If you're writing such a library, you have have two options:
+## How to test libraries on .NET Core
 
-1. Reference the native `.dll` directly in your `project.json`.
-2. Package that `.dll` into its own NuGet package and depend on that package.
+It's important to be able to test across platforms.  You can use either [xUnit](http://xunit.github.io/) or MSTest out of the box.  Both either are perfectly suitable for unit testing your library on .NET Core.  How you set up your solution with test projects will depend on the [structure of your solution](#structuring-a-solution).  The following example assumes that test and source directories live in the same top-level directory.
 
-For the first option, you'll need to include the following in your `project.json` file:
+1. Set up your solution.  You can do so with the following commands:
 
-1. Setting `allowUnsafe` to `true` in a `buildOptions` section.
-2. Specifying a [Runtime Identifier (RID)](../rid-catalog.md) in a `runtimes` section.
-3. Specifying the path to the native `.dll` file(s) that you are referencing.
+```bash
```

Comment:
Consider links to the dotnet commands near this example (https://github.com/dotnet/docs/tree/master/docs/core/tools), specifically `dotnet new` and `dotnet sln`